packages/core/ios/RNEmbrace/EmbraceManager.swift

```diff
@@ -0,0 +1,301 @@
+    }
+  }
+  
+  @objc
+  func setReactNativeVersion(_ version: String, resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
+    do {
+      try Embrace.client?.metadata.addResource(key: EmbraceKeys.reactNativeVersion.rawValue, value: version)
+      resolve(true)
+    } catch let error {
+      reject("SET_RECT_NATIVE_VERSION", "Error setting React Native Number", error)
+    }
+  }
+  
+  @objc
+  func removeSessionProperty(_ key: String, resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
+    do {
+      // TODO REfactor to include lifespan
+      try Embrace.client?.metadata.removeProperty(key: key)
+      resolve(true)
+    } catch let error {
+      reject("REMOVE_SESSION_PROPERTY", "Error removing Session Property", error)
+    }
+  }
+  
+  @objc
+  func setUserEmail(_ userEmail: String, resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
+    Embrace.client?.metadata.userEmail = userEmail
+    resolve(true)
+  }
+  
+  @objc
+  func addSessionProperty(
+    _ key: String,
+    value: String,
+    permanent: Bool,
+    resolver resolve: @escaping RCTPromiseResolveBlock,
+    rejecter reject: @escaping RCTPromiseRejectBlock
+  ) {
+    do {
+      let lifespan:MetadataLifespan = permanent ? .permanent : .session
+      try Embrace.client?.metadata.addProperty(key:key, value:value, lifespan:lifespan)
+      resolve(true)
+    } catch let error {
+      reject("ADD_SESSION_PROPERTY", "Error adding Session Property", error)
+    }
+  }
+  
+  @objc(clearUserPersona:resolver:rejecter:)
+  func clearUserPersona(_ persona:String, resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
+    do {
+      try Embrace.client?.metadata.remove(persona: PersonaTag(persona), lifespan: .session)
+      resolve(true)
+    }catch let error {
+      reject("CLEAR_USER_PERSONA", "Error removing User Persona", error)
+    }
+  }
+  
+  @objc
+  func logMessageWithSeverityAndProperties(
+    _ message: String,
+    severity: String,
+    properties: NSDictionary,
+    resolver resolve: @escaping RCTPromiseResolveBlock,
+    rejecter reject: @escaping RCTPromiseRejectBlock
+  ) {
+    
+    let severityValue = self.severityFromString(from:severity)
+    guard let attributes = properties as? [String: String] else {
+      reject("LOG_MESSAGE_INVALID_PROPERTIES", "Properties should be [String: String]", nil)
+      return
+    }
+    Embrace.client?.log(
+      message,
+      severity: severityValue,
+      attributes: attributes
+    )
+    resolve(true)
+    
+  }
+  
+  private func severityFromString(from inputString: String) -> LogSeverity {
+    switch inputString {
+    case "info":
+      return .info
+    case "warning":
+      return .warn
+    default:
+      return .error
+    }
+  }
+}
```

Comment:
for this file I'd suggest reorganizing the methods so that related ones are together, I'd say we can just follow how the table is laid out in: https://www.notion.so/embraceio/Deprecation-of-Embrace-specific-methods-in-the-RN-API-in-favour-of-OTEL-equivalents-62a6fa6d3e344490809b156fad2938e7?pvs=4 with comment blocks separating the categories like "Utility" and "Session"